File: src/main/java/hacker/solution/TwoDArray.java
package hacker.solution;

import java.util.Scanner;

public class TwoDArray {
    private static final Scanner scanner = new Scanner(System.in);
    public static void main(String[] args) {
        int[][] arr = new int[6][6];

        for (int i = 0; i < 6; i++) {
            String[] arrRowItems = scanner.nextLine().split(" ");
            scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

            for (int j = 0; j < 6; j++) {
                int arrItem = Integer.parseInt(arrRowItems[j]);
                arr[i][j] = arrItem;
            }
        }

        scanner.close();
        int bestSum = -100000;
        for (int r=0;r<4;r++) {
            for (int c=0;c<4;c++) {
                int[][] h = getHourglass(arr,r,c);
                int val = getHourglassSum(h);
                if (val > bestSum) bestSum = val;
            }
        }
        System.out.println(bestSum);
    }
    private static int getHourglassSum(int[][] a) {
        int sum = 0;
        for (int row=0;row<3;row++) {
           if (row==1) {
                sum += a[row][1];
            } else {
                for (int col = 0; col < 3; col++) {
                    sum += a[row][col];
                }
            }
        }
        return sum;
    }
    private static int[][] getHourglass(int[][] a, int r, int c) {
        int[][] b = new int[3][3];
        for (int row=0;row<3;row++) {
            for (int col=0;col<3;col++) {
                b[row][col] = a[row+r][col+c];
            }
        }
        return b;
    }
}

File: src/main/kotlin/hacker/solutionk/TwoDArray.kt
package hacker.solutionk

import java.util.*

object TwoDArray {
    private val scanner = Scanner(System.`in`)
    @JvmStatic
    fun main(args: Array<String>) {
        val arr = Array(6) { IntArray(6) }

        for (i in 0..5) {
            val arrRowItems = scanner.nextLine().split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?")

            for (j in 0..5) {
                val arrItem = Integer.parseInt(arrRowItems[j])
                arr[i][j] = arrItem
            }
        }

        scanner.close()
        var bestSum = -100000
        for (r in 0..3) {
            for (c in 0..3) {
                val h = getHourglass(arr, r, c)
                val `val` = getHourglassSum(h)
                if (`val` > bestSum) bestSum = `val`
            }
        }
        println(bestSum)
    }

    private fun getHourglassSum(a: Array<IntArray>): Int {
        var sum = 0
        for (row in 0..2) {
            if (row == 1) {
                sum += a[row][1]
            } else {
                for (col in 0..2) {
                    sum += a[row][col]
                }
            }
        }
        return sum
    }

    private fun getHourglass(a: Array<IntArray>, r: Int, c: Int): Array<IntArray> {
        val b = Array(3) { IntArray(3) }
        for (row in 0..2) {
            for (col in 0..2) {
                b[row][col] = a[row + r][col + c]
            }
        }
        return b
    }
}
